src/feed/api.ts

```diff
@@ -64,23 +72,51 @@ export async function writeFeedData(
  * @param connection connection information for data uploading
  * @param topic key for data
  * @param data data to upload
- * @param privateKey private key to sign data
+ * @param wallet feed owner's wallet
  * @param epoch feed epoch
  */
 export async function writeFeedDataRaw(
   connection: Connection,
   topic: string,
   data: Uint8Array,
-  privateKey: string | Uint8Array,
+  wallet: utils.HDNode,
   epoch?: Epoch,
 ): Promise<Reference> {
   if (!epoch) {
-    epoch = new Epoch(HIGHEST_LEVEL, getUnixTimestamp())
+    epoch = await getLastEpoch(connection.bee, topic, prepareEthAddress(wallet.address))
   }
 
   const topicHash = bmtHashString(topic)
   const id = getId(topicHash, epoch.time, epoch.level)
-  const socWriter = connection.bee.makeSOCWriter(privateKey)
+  const socWriter = connection.bee.makeSOCWriter(wallet.privateKey)
 
   return socWriter.upload(connection.postageBatchId, id, data)
 }
+
+/**
+ * Gets last epoch for `topic` using owner's `address`
+ */
+export async function getLastEpoch(bee: Bee, topic: string, address: Utils.EthAddress | Uint8Array): Promise<Epoch> {
+  let epoch = new Epoch(HIGHEST_LEVEL, getUnixTimestamp())
+  try {
+    const feedData = await getFeedData(bee, topic, address)
+    feedData.epoch.level = feedData.epoch.getNextLevel(feedData.epoch.time)
+    epoch = feedData.epoch
+    // eslint-disable-next-line no-empty
+  } catch (e) {}
+
+  return epoch
+}
+
+/**
+ * Deletes feed data for `topic` using owner's `wallet`
+ */
+export async function deleteFeedData(
+  connection: Connection,
+  topic: string,
+  wallet: utils.HDNode,
+  podPassword: PodPasswordBytes,
+  epoch?: Epoch,
+): Promise<Reference> {
+  return writeFeedData(connection, topic, stringToBytes(DELETE_FEED_MAGIC_WORD), wallet, podPassword, epoch)
```

Comment:
really wrong practice.
this will not replace the data under the SOC address just put it into an undefined behaviour where either of the payload can be retreived from the network.